docs/user_guide/building_blocks/config_objects.md

```diff
@@ -53,9 +125,374 @@ given via the ```--variables``` argument, accessing information from columns bas
+```python
+cfg.x.external_files = DotDict.wrap({
+    # lumi files
+    "lumi": {
+        "golden": ("/afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/Legacy_2017/Cert_294927-306462_13TeV_UL2017_Collisions17_GoldenJSON.txt", "v1"),  # noqa
+        "normtag": ("/afs/cern.ch/user/l/lumipro/public/Normtags/normtag_PHYSICS.json", "v1"),
+    },
+
+    # muon scale factors
+    "muon_sf": (f"{json_mirror}/POG/MUO/2017_UL/muon_Z.json.gz", "v1"),
+
+    # some example of a ML training dataset to show an example with URL
+    "vehicules_dataset_fastai": ("https://github.com/arunoda/fastai-courses/releases/download/fastai-vehicles-dataset/fastai-vehicles.tgz", "v1"),  # noqa
+})
+```
+
+An example of usage of the `muon_sf`, including the requirement of the BundleExternalFiles task is given in the {py:mod}`~columnflow.production.cms.muon` file, implementing the {py:class}`~columnflow.production.cms.muon.muon_weights` Producer.
+
+TODO: put the following in drop down menu
+
+Showing how to require the BundleExternalFiles task to have run in the example of the muon weights Producer linked above:
+```python
+@muon_weights.requires
+def muon_weights_requires(self: Producer, reqs: dict) -> None:
+    if "external_files" in reqs:
+        return
+
+    from columnflow.tasks.external import BundleExternalFiles
+    reqs["external_files"] = BundleExternalFiles.req(self.task)
+```
+
+
+### Luminosity
+
+The luminosity, needed for some normalizations and for the labels in the standard columnflow plots, needs to be given in the auxiliary arguments `cfg.x.luminosity` as an object of the {external+scinum:py:class}`scinum.Number` class, such that for example the `nominal` parameter exists.
+An example for a CMS luminosity of 2017 with uncertainty sources given as relative errors is given below.
+
+```python
+from scinum import Number
+cfg.x.luminosity = Number(41480, {
+    "lumi_13TeV_2017": 0.02j,
+    "lumi_13TeV_1718": 0.006j,
+    "lumi_13TeV_correlated": 0.009j,
+})
+```
+
+
+### Defaults
+
+Default values can be given for several command line parameters in columnflow, using the `cfg.x.default_{parameter}` entry in the Config object.
+The expected format is a single string containing the name of the object to be used as default for parameters accepting only one argument and a tuple for parameters accepting several arguments.
+The command-line arguments supporting a default value from the Config object are given in the cms_minimal example of the analysis_templates and shown again below:
+
+```python
+cfg.x.default_calibrator = "example"
+cfg.x.default_selector = "example"
+cfg.x.default_producer = "example"
+cfg.x.default_ml_model = None
+cfg.x.default_inference_model = "example"
+cfg.x.default_categories = ("incl",)
+cfg.x.default_variables = ("n_jet", "jet1_pt")
+cfg.x.default_weight_producer = "example"
+```
```

Comment:
Would rather import this from the example analysis directly

Reply:
I added the `cfg.x.default_weight_producer = "example"` inside, which doesn't exist in the current master branch (still in pull request) and was not added to the template in the corresponding pull request, should it just be ignored for now?

Reply:
done, removed